### ISSUE TITLE
[BUG] Compilter complains about set but unused variable during build

### ISSUE THREAD

**Issue by sempervictus** [penguintrace/penguintrace#2]
**Describe the bug**
When building on Arch Linux (`clang version 14.0.6`/llvm package v`14.0.6-3`), the build fails due to:
```
  [ CXX] dwarf/DIEPrint.cpp
dwarf/DIEPrint.cpp:282:12: error: variable 'store' set but not used [-Werror,-Wunused-but-set-variable]
      DIE* store = nullptr;
           ^
1 error generated.
make: *** [Makefile:121: build/dwarf/DIEPrint.o] Error 1
```

**To Reproduce**
Steps to reproduce the behaviour:
1. Pull sources on Arch Linux
2. Install dependencies
3. Clear build flags to get around architecure targeting and LTO related build problems:
```
  unset CFLAGS
  unset CXXFLAGS
  unset LDFLAGS
  unset LTOPLUGIN
  unset ARFLAGS
  unset RANLIBFLAGS
  unset NMFLAGS
```
4. `make`

**Expected behaviour**
Build succeeds

---

**Additional context**
Attempted to mitigate by `CXXFLAGS="-Wno-unused-but-set-variable -Wno-unused-variable -Wno-error=unused-variable" make` invocation to no avail

**Comment by martiansoup** [comment]
I think this should now be resolved by commit 08dedd09c7320d9ce70adc806f5bc0f667d68fbb

**Comment by sempervictus** [comment]
Thank you, that problem is resolved:
```
  [ CXX] dwarf/DIE.cpp
  [ CXX] dwarf/DIEPrint.cpp
  [ CXX] dwarf/DIETraversal.cpp
  [ CXX] dwarf/DIEValue.cpp
```
now on to
```
...
  [ CXX] dwarf/DIEValue.cpp
  [ CXX] dwarf/Frame.cpp
In file included from dwarf/Frame.cpp:23:
dwarf/Frame.h:214:14: error: no template named 'unique_ptr' in namespace 'std'
        std::unique_ptr<std::map<std::string, RegRule> > unwindInfo(uint64_t pc);
        ~~~~~^
```
:smile:
Thank you
